src/probnum/diffeq/perturbedsolvers/_perturbation_functions.py

```diff
@@ -0,0 +1,50 @@
+"""perturbation functions to perturb the stepsize."""
+import numpy as np
+
+
+def perturb_uniform(step, order, noise_scale, seed=1):
+    """perturbs the step with uniformly distributed noise scaled by noise-scale.
+    proposed by Abdulle and Garegnani(2020)
+
+    Parameters
+    ----------
+    step : float
+        unperturbed step propesed by the steprule
+    order : int
+        order of the solver
+    noise_scale : float
+        scales the perturbation
+    seed : int
+        seed for pseudo-random number generator.
+    """
+    np.random.seed(seed=seed)
+    if step < 1:
+        noisy_step = np.random.uniform(
+            step - noise_scale * step ** (order + 0.5),
+            step + noise_scale * step ** (order + 0.5),
+        )
+    else:
+        print("Error: Stepsize too large (>=1), not possible")
```

Comment:
Instead of printing "error", I think it would be better to either raise a "ValueError" or at least throw a "real" warning (https://docs.python.org/3/library/warnings.html). Since the implementation actually breaks here, I would prefer the ValueError though.